### PR TITLE
ENH: Add numba engine to rolling/expanding.std/var

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/tests/window/test_numba.py [hunk 1]
```diff
@@ -282,19 +286,24 @@ def f(x):
...
-        )(engine_kwargs=engine_kwargs, engine="numba")
-        expected = getattr(
-            df.rolling(2, method="single", axis=axis, min_periods=0), method
-        )(engine_kwargs=engine_kwargs, engine="numba")
-        tm.assert_frame_equal(result, expected)
+        if method in ("var", "std"):
+            with pytest.raises(NotImplementedError, match=f"{method} not supported"):
+                getattr(
+                    df.rolling(2, method="table", axis=axis, min_periods=0), method
+                )(engine_kwargs=engine_kwargs, engine="numba", **kwargs)
```
Comment (jbrockmendel) [inline pandas/tests/window/test_numba.py:298]: can these big calls be split into multiple lines?

**Comment by mroeschke** [comment]
Failure due to codecov not being able to tell the numba decorated function is inherently tested.

File: doc/source/whatsnew/v1.4.0.rst [hunk 2]
```diff
@@ -212,7 +212,8 @@ Other enhancements
 - :meth:`.GroupBy.mean` now supports `Numba <http://numba.pydata.org/>`_ execution with the ``engine`` keyword (:issue:`43731`)
 - :meth:`Timestamp.isoformat`, now handles the ``timespec`` argument from the base :class:``datetime`` class (:issue:`26131`)
 - :meth:`NaT.to_numpy` ``dtype`` argument is now respected, so ``np.timedelta64`` can be returned (:issue:`44460`)
--
+- :meth:`.Rolling.var`, :meth:`.Expanding.var`, :meth:`.Rolling.std`, :meth:`.Expanding.std` now support `Numba <http://numba.pydata.org/>`_ execution with the ``engine`` keyword (:issue:`44461`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.4.0.rst:215]: at some point should group these doc-string changes together (maybe into a separate section ok too but not requried)

File: pandas/core/window/rolling.py [hunk 3]
```diff
@@ -1459,8 +1460,24 @@ def median(
...
+        ddof: int = 1,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict[str, bool] | None = None,
+        **kwargs,
+    ):
         nv.validate_window_func("std", args, kwargs)
+        if maybe_use_numba(engine):
+            if self.method == "table":
+                raise NotImplementedError("std not supported with method='table'")
```
Comment (jreback) [inline pandas/core/window/rolling.py:1474]: hit this in tests?

File: pandas/core/window/rolling.py [hunk 4]
```diff
@@ -1472,8 +1489,24 @@ def zsqrt_func(values, begin, end, min_periods):
...
+        ddof: int = 1,
+        *args,
+        engine: str | None = None,
+        engine_kwargs: dict[str, bool] | None = None,
+        **kwargs,
+    ):
         nv.validate_window_func("var", args, kwargs)
+        if maybe_use_numba(engine):
+            if self.method == "table":
+                raise NotImplementedError("var not supported with method='table'")
```
Comment (jreback) [inline pandas/core/window/rolling.py:1503]: same question

**Review by jreback — CHANGES_REQUESTED** [review]
is the arm failure real?

**Comment by mroeschke** [comment]
> is the arm failure real?

It's this flaky test

```
@pytest.mark.single
    def test_unique_bad_unicode(index_or_series):
        # regression test for #34550
        uval = "\ud83d"  # smiley emoj
```

File: pandas/core/window/rolling.py (hunk 3 above)
Comment (mroeschke) [reply pandas/core/window/rolling.py:1474]: Yup here: https://github.com/pandas-dev/pandas/pull/44461/files#diff-4fea2b6ab86ef8bb3c962ab4ee5667c2ea4df13bfab32ffcf03ec130451699dcR296

File: pandas/core/window/rolling.py (hunk 4 above)
Comment (mroeschke) [reply pandas/core/window/rolling.py:1503]: Yup here: https://github.com/pandas-dev/pandas/pull/44461/files#diff-4fea2b6ab86ef8bb3c962ab4ee5667c2ea4df13bfab32ffcf03ec130451699dcR296

File: doc/source/whatsnew/v1.4.0.rst (hunk 2 above)
Comment (mroeschke) [reply doc/source/whatsnew/v1.4.0.rst:215]: Yeah once I add median I can group these better

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**